File: ecommerce-platform-frontend/src/app/dto/tag/response-tag-dto.ts
import {ResponseProductDTO} from '../product/response-product-dto';
import {ResponseCategoryDTO} from '../category/response-category-dto';
import {ResponseMixtureDTO} from '../mixtures/response-mixture-dto';
import {BaseUpdateOrResponseDTO} from '../base/base-update-or-response.dto';

export interface ResponseTagDTO extends BaseUpdateOrResponseDTO {
  categories?: ResponseCategoryDTO[];
  products?: ResponseProductDTO[];
  mixtures?: ResponseMixtureDTO[];
}

File: catalog-service/src/main/java/com/dvoracekmartin/catalogservice/application/dto/tag/ResponseTagDTO.java
package com.dvoracekmartin.catalogservice.application.dto.tag;

import com.dvoracekmartin.catalogservice.application.dto.base.BaseUpdateOrResponseDTO;
import com.dvoracekmartin.catalogservice.application.dto.category.ResponseCategoryDTO;
import com.dvoracekmartin.catalogservice.application.dto.media.MediaDTO;
import com.dvoracekmartin.catalogservice.application.dto.mixture.ResponseMixtureDTO;
import com.dvoracekmartin.catalogservice.application.dto.product.ResponseProductDTO;
import lombok.Data;

import java.util.List;

@Data
public class ResponseTagDTO extends BaseUpdateOrResponseDTO {

    List<ResponseCategoryDTO> categories;
    List<ResponseProductDTO> products;
    List<ResponseMixtureDTO> mixtures;

    public ResponseTagDTO(Long id, String name, String description, int priority, boolean active, List<MediaDTO> media, List<ResponseCategoryDTO> categories, List<ResponseProductDTO> products, List<ResponseMixtureDTO> mixtures) {
        super(id, name, description, priority, active, media);
        this.categories = categories;
        this.products = products;
        this.mixtures = mixtures;
    }
}
